domain/macaroon/state/rootkeys.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+	"time"
+
+	"github.com/canonical/sqlair"
+	"github.com/go-macaroon-bakery/macaroon-bakery/v3/bakery/dbrootkeystore"
```

Comment:
This shouldn't leak into state.

Reply:
Let's get together to discuss this. I agree _somewhat_. We exchange core types in and out of state which is similar. 

The most important thing is that we're not storing these types directly like we were with Mongo. So although a change upstream would require Juju changes, we don't end up with potential data inconsistencies like we could before.

Reply:
> We exchange core types in and out of state which is similar.

Those are types that Juju owns, in that I agree. For third-party libraries, let us add a tale of caution.